internal/test/test_test.go

```diff
@@ -209,11 +209,95 @@ func TestExecutingTests(t *testing.T) {
 		testFiles := map[string][]byte{
 			script.Filename: script.Source,
 		}
-		results, coverageReport, err := testCode(testFiles, state, true)
+		flags := flagsTests{
+			Cover: true,
+		}
+		results, coverageReport, err := testCode(testFiles, state, flags)
 
 		require.NoError(t, err)
-		require.Len(t, results, 1)
-		assert.NoError(t, results[script.Filename][0].Error)
+		require.Len(t, results[script.Filename], 3)
+		for _, result := range results[script.Filename] {
+			assert.NoError(t, result.Error)
+		}
+
+		location := common.StringLocation("FooContract.cdc")
+		coverage := coverageReport.Coverage[location]
+
+		assert.Equal(t, []int{}, coverage.MissedLines())
+		assert.Equal(t, 15, coverage.Statements)
+		assert.Equal(t, "100.0%", coverage.Percentage())
+		assert.EqualValues(
+			t,
+			map[int]int{
+				6: 1, 14: 1, 18: 10, 19: 1, 20: 9, 21: 1, 22: 8, 23: 1,
+				24: 7, 25: 1, 26: 6, 27: 1, 30: 5, 31: 4, 34: 1,
+			},
+			coverage.LineHits,
+		)
+
+		assert.True(t, coverageReport.TotalLocations() > 1)
+		assert.ElementsMatch(
+			t,
+			[]string{
+				"s.7465737400000000000000000000000000000000000000000000000000000000",
+				"I.Crypto",
+				"I.Test",
+				"A.0ae53cb6e3f42a79.FlowToken",
+				"A.f8d6e0586b0a20c7.FlowStorageFees",
+				"A.f8d6e0586b0a20c7.FlowDKG",
+				"A.f8d6e0586b0a20c7.ExampleNFT",
+				"A.f8d6e0586b0a20c7.NonFungibleToken",
+				"A.f8d6e0586b0a20c7.FlowIDTableStaking",
+				"A.f8d6e0586b0a20c7.FlowClusterQC",
+				"A.f8d6e0586b0a20c7.NodeVersionBeacon",
+				"A.f8d6e0586b0a20c7.StakingProxy",
+				"A.f8d6e0586b0a20c7.FUSD",
+				"A.e5a8b7f23e8b548f.FlowFees",
+				"A.ee82856bf20e2aa6.FungibleToken",
+				"A.f8d6e0586b0a20c7.FlowStakingCollection",
+				"A.f8d6e0586b0a20c7.MetadataViews",
+				"A.f8d6e0586b0a20c7.NFTStorefrontV2",
+				"A.f8d6e0586b0a20c7.NFTStorefront",
+				"A.f8d6e0586b0a20c7.LockedTokens",
+				"A.f8d6e0586b0a20c7.FlowServiceAccount",
+				"A.f8d6e0586b0a20c7.FlowEpoch",
+			},
+			coverageReport.ExcludedLocationIDs(),
+		)
+		assert.Equal(
+			t,
+			"Coverage: 97.2% of statements",
+			coverageReport.String(),
+		)
+	})
+
+	t.Run("with code coverage for contracts only", func(t *testing.T) {
+		t.Parallel()
+
+		// Setup
+		_, state, _ := util.TestMocks(t)
+
+		state.Contracts().AddOrUpdate(config.Contract{
+			Name:     tests.ContractFooCoverage.Name,
+			Location: tests.ContractFooCoverage.Filename,
+		})
+
+		// Execute script
+		script := tests.TestScriptWithCoverage
+		testFiles := map[string][]byte{
+			script.Filename: script.Source,
+		}
+		flags := flagsTests{
+			Cover:     true,
+			CoverCode: contractsCoverCode,
```

Comment:
A question in the struct description there is '"default:"all" flag:"covercode"' for CoverCode string.
But const contractsCoverCode is equal to "contracts". So "covercode" is what a user has to specify and thens somewhere it is converted to contractsCoverCode value ?

Reply:
The idea is that you add a default value to a flag, for the most used scenario. But developers can provide a different value to the flag. For example:

```bash
flow test --cover --covercode="contracts" tests/test_approval_voting.cdc
```

The `const contractsCoverCode` is added just as a best practice, to have magic strings defined in a single place, to avoid typos and having to update multiple places. So, if I want to update its value to something else, I can do it in a single place, in the `const` declaration.

Reply:
Valid point about listing the available values for this flag :ok_hand: 
Updated in https://github.com/onflow/flow-cli/pull/1074/commits/3debe6f2a669c967c3a6e246b1cb4526a559d972